app/parse.py

```diff
@@ -15,8 +31,108 @@ class Product:
     num_of_reviews: int
 
 
+def parse_product(product_element: WebElement) -> Product:
+    return Product(
+        title=product_element.find_element(
+            By.CLASS_NAME, "title"
+        ).get_attribute("title"),
+        description=product_element.find_element(
+            By.CLASS_NAME, "description"
+        ).text,
+        price=float(
+            product_element.find_element(
+                By.CLASS_NAME, "price"
+            ).text.replace("$", "")
+        ),
+        rating=len(
+            product_element.find_elements(
+                By.CLASS_NAME, "ws-icon-star"
+            )
+        ),
+        num_of_reviews=int(
+            product_element.find_element(
+                By.CLASS_NAME, "review-count"
+            ).text.split()[0]
+        ),
+    )
+
+
+def get_all_products_from_page(page_url: str) -> list[Product]:
+    with webdriver.Chrome() as driver:
+        driver.get(page_url)
+
+        try:
+            cookie_banner = WebDriverWait(driver, 10).until(
+                ec.visibility_of_element_located((By.ID, "cookieBanner"))
+            )
+            if cookie_banner:
+                close_button = cookie_banner.find_element(
+                    By.ID, "closeCookieBanner"
+                )
+                close_button.click()
+        except Exception as e:
+            print("Couldn't close Cookie Banner:", e)
+
+        try:
+            while True:
+                more_buttons = driver.find_elements(
+                    By.CLASS_NAME, "ecomerce-items-scroll-more"
+                )
+                if not more_buttons:
+                    break
+
+                more_button = more_buttons[0]
+                if more_button.value_of_css_property("display") == "none":
+                    break
+
+                WebDriverWait(driver, 10).until(
+                    ec.element_to_be_clickable(more_button)
+                )
+                more_button.click()
+        except Exception as e:
+            print("Exception:", e)
+
+        product_elements = driver.find_elements(
+            By.CLASS_NAME, "product-wrapper"
+        )
+
+        return [
+            parse_product(product_element)
+            for product_element in product_elements
+        ]
+
+
 def get_all_products() -> None:
-    pass
+    products = {
+        "home.csv": get_all_products_from_page(HOME_URL),
+        "computers.csv": get_all_products_from_page(COMPUTERS_URL),
+        "laptops.csv": get_all_products_from_page(LAPTOPS_URL),
+        "tablets.csv": get_all_products_from_page(TABLETS_URL),
+        "phones.csv": get_all_products_from_page(PHONES_URL),
+        "touch.csv": get_all_products_from_page(TOUCH_URL)
+    }
+
+    write_products_to_csv(products)
+
+
+def write_products_to_csv(data: dict[str, list[Product]]) -> None:
+    for csv_file_name, products_list in data.items():
+        with open(csv_file_name, "w", newline="", encoding="utf-8") as file:
```

Comment:
Consider adding exception handling around file operations to manage potential I/O errors, such as file permission issues or disk space limitations, which could occur during the CSV writing process.